samples/subsys/zbus/benchmark/src/sub.c

```diff
@@ -0,0 +1,65 @@
+
+extern atomic_t count;
+
+ZBUS_CHAN_DECLARE(bm_channel);
+
+#define SFY(i, _) s##i
+
+#define CONSUMERS_NAME_LIST LISTIFY(CONFIG_BM_ONE_TO, SFY, (, /* separator */))
+
+#define CREATE_OBSERVER(s) ZBUS_SUBSCRIBER_DEFINE(s, 4)
+
+FOR_EACH(CREATE_OBSERVER, (;), CONSUMERS_NAME_LIST)
+	;
+
+#define CREATE_OBSERVATIONS(s) ZBUS_CHAN_ADD_OBS(bm_channel, s, 3)
+
+FOR_EACH(CREATE_OBSERVATIONS, (;), CONSUMERS_NAME_LIST)
+	;
+
+#define S_TASK(name)                                                                               \
+	void name##_task(void)                                                                     \
+	{                                                                                          \
+		const struct zbus_channel *chan;                                                   \
+                                                                                                   \
+		while (1) {                                                                        \
+			if (zbus_sub_wait(&name, &chan, K_FOREVER) == 0) {                         \
+				if (zbus_chan_claim(chan, K_FOREVER) != 0) {                       \
+					k_oops();                                                  \
+				}                                                                  \
+				if (IS_ENABLED(CONFIG_BM_FAIRPLAY)) {                              \
+					struct bm_msg message;                                     \
+                                                                                                   \
+					memcpy(zbus_chan_msg(chan), &message, chan->message_size); \
+                                                                                                   \
+					atomic_add(&count, *((uint16_t *)message.bytes));          \
+				} else {                                                           \
+					const struct bm_msg *msg_received =                        \
+						zbus_chan_const_msg(chan);                         \
+                                                                                                   \
+					atomic_add(&count, *((uint16_t *)msg_received->bytes));    \
+				}                                                                  \
+				zbus_chan_finish(chan);                                            \
+			} else {                                                                   \
+				k_oops();                                                          \
+			}                                                                          \
+		}                                                                                  \
+	}                                                                                          \
+                                                                                                   \
+	K_THREAD_DEFINE(name##_id, CONSUMER_STACK_SIZE, name##_task, NULL, NULL, NULL, 3, 0, 0);
+
+#define CREATE_TASKS(s) S_TASK(s)
+FOR_EACH(CREATE_TASKS, (;), CONSUMERS_NAME_LIST)
+	;
```

Comment:
```suggestion
FOR_EACH(CREATE_TASKS, (;), CONSUMERS_NAME_LIST);
```
? For all of these I guess.

Reply:
yes please, let's not arbitrarily accept `clang-format` output, it's known to screw up macros, if you want to be able to rerun it just add a pair of clang format guards:

```
/* clang-format off */
..
/* clang-format on */
```